hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -302,6 +303,7 @@ class HoodieSparkSqlWriterInternal {
           else KeyGeneratorType.getKeyGeneratorClassName(hoodieConfig)
         HoodieTableMetaClient.newTableBuilder()
           .setTableType(tableType)
+          .setTableVersion(Integer.valueOf(getStringWithAltKeys(parameters, HoodieWriteConfig.WRITE_TABLE_VERSION)))
```

Comment:
Isn't table version and write table version two different config options, not sure why setting up the table version always from the write table version, the write table version is auxiliary for migration, I kind of think we should not set it up explicitly every time we initiate the table. can we detect the table version automically from the table config 
 and validate against the write table version then?